source/Halibut/Transport/Observability/ObservableStream.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Halibut.Transport.Observability
+{
+    public class ObservableStream : Stream
```

Comment:
Is `ByteCountingStream` a better name for this stream?

Reply:
Yeah, done 😁